crates/daphne-server/src/lib.rs

```diff
@@ -100,6 +112,45 @@ impl router::DaphneService for App {
     fn signing_key(&self) -> Option<&p256::ecdsa::SigningKey> {
         self.service_config.signing_key.as_ref()
     }
+
+    async fn check_bearer_token(
+        &self,
+        token: &BearerToken,
+        sender: DapSender,
+        task_id: Option<TaskId>,
+    ) -> Result<(), String> {
+        match (&self.service_config.peer_auth, sender, task_id) {
```

Comment:
This `service_config.peer_auth` is specific to taskprov, correct? If so, it should be named accordingly.

Reply:
it is not specific to taskprov no. This is just a bearer token we accept

Reply:
I believe this is necessary because it avoids granting the "taskprov peer" access to tasks it hasn't configured.